server/src/main/java/org/elasticsearch/search/rank/RankContextInternal.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.Rewriteable;
+import org.elasticsearch.index.query.SearchExecutionContext;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Holds internal information about the queries
+ * used for ranking required for shard phases.
+ */
+public class RankContextInternal implements Writeable, Rewriteable<RankContextInternal> {
```

Comment:
I agree. I initially thought it would be more complex so that additional abstraction would be useful. I have removed the abstraction in favor of your suggestion here. The only thing I didn't do was add an additional method to SearchExecutionContext because it returns ParsedQuery for its toQuery methods, so I thought that might be confusing. I just added a simple loop where we do the conversion instead.